### PR TITLE
Use malloc to avoid alignment problems.

### PR DESCRIPTION
**Pull request by richsalz**
Thanks to Norm Green for reporting this issue.

This PR is based on code @vdukhovni wrote and posted.

### CODE REVIEW

**Comment by dot-asm** [comment]
I wonder which platform has got the problem and how come it went unnoticed so far. On some level I rather wonder if it's outcome of some recent compiler change that misaligns stack, and in such case how far would it go. [Do note that I'm not saying that suggested fix is not appropriate, only wonder how come it worked so far. Because customarily objects on stack would be aligned at pointer size, even if their declaration doesn't imply corresponding guarantee.]

**Comment by levitte** [comment]
@dot-asm, it's Solaris on SPARC (maybe Solaris64).  See [this thread](https://mta.openssl.org/pipermail/openssl-users/2018-February/007549.html) on openssl-users.

File: test/asn1_encode_test.c [hunk 1]
```diff
@@ -709,15 +709,18 @@ static int do_encode_custom(EXPECTED *input,
...
-    ASN1_VALUE *o = (ASN1_VALUE *)&buf;
+    ASN1_VALUE *o;
     int ret;
 
     OPENSSL_assert(package->encode_expectations_elem_size <= DATA_BUF_SIZE);
+    if ((o = OPENSSL_malloc(DATA_BUF_SIZE)) == NULL)
+        return 0;
 
-    (void)RAND_bytes(buf, (int)package->encode_expectations_elem_size);
+    (void)RAND_bytes((unsigned char*)o,
```
Comment (bernd-edlinger) [inline test/asn1_encode_test.c]: I wonder if the result code of RAND_bytes should be checked?

Comment (richsalz) [reply test/asn1_encode_test.c]: RAND is just a side-effect of the test, and this PR just fixes the alignment issue.  Perhaps a new PR to go check all uses and fix them?

(Besides, I'm about to push this :)

Comment (vdukhovni) [reply test/asn1_encode_test.c]: There's no need to check the output of RAND_bytes() here. We don't care what data is placed in "buf", that's why RAND_bytes is used.  We could equally memset() the buffer to zero. We're testing the templates, not the data.

**Comment by richsalz** [comment]
Merged to master, thanks!

**Pull request closed without merging**